querybook/webapp/components/DataDocViewersBadge/DataDocViewersBadge.tsx

```diff
@@ -131,7 +131,12 @@ export const DataDocViewersBadge = React.memo<IDataDocViewersBadgeProps>(
                             uid: viewerInfo.uid,
                             tooltip:
                                 viewerInfo.uid in userInfoById
-                                    ? userInfoById[viewerInfo.uid].username
+                                    ? (userInfoById[viewerInfo.uid].fullname ??
+                                          userInfoById[viewerInfo.uid]
+                                              .username) +
+                                      (userInfoById[viewerInfo.uid].deleted
+                                          ? ' (deactivated)'
+                                          : '')
```

Comment:
can we have separate function for it? also it's not very readable as it has nested ternary operations to me.

btw for the elastics search, you have `"fullname": (fullname or username) + (" (deactivated)" if user.deleted else "")` on the backend, while this one is handled on frontend, and also seeing something similar below like `@{userName} {deletedText}` (UserBadge.tsx)

Would it be better to make the fullname transform either on the backend and can have a shared function to transformation, or it should be on the frontend, so the string of ` (deactivated)` could be more customizable?

Reply:
there are 2 sources of data:
- db
- elasticsearch
so i am adding `(deactivated)` for data from db. And for elasticsearch i am adding it to the full name field. Ideally, we should add a `deleted` field in elasticsearch so we don't have to do any transformation in the backend. However, i think we can do that later since it is somewhat of a breaking change